### PR TITLE
Support for directly pulling IIIF resources from content servers

### PR DESCRIPTION
**Pull request by singaltanmay**
# Description

Uses the IIIF `resourceUrl` metadata value in the `IIIF` domain to directly pull image resources from IIIF content servers. [OpenSeaDragon](https://openseadragon.github.io/) image viewer is used to provide an easy-to-use interface for the users.

IIIF support has not yet been merged into cineast (see https://github.com/vitrivr/cineast/pull/217) so in order to make use of these new features you would need to checkout a compatible version of Cineast from https://github.com/singaltanmay/cineast/tree/feature/IIIF_Presentation_API and run an IIIF extraction job

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have reviewed my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (wiki, README, Website)
- [ ] I have updated the pre-built binary on the release page (for major changes)

### CODE REVIEW

**Review by silvanheller — COMMENTED** [review]
This looks good to me, thanks for the work @singaltanmay! I'll rely on the verdict of @Spiess since I assume he'll be doing a full-stack test.

**Review by Spiess — APPROVED** [review]
I have tested the changes with extracted IIIF resources, regular locally hosted images, image sequences and videos.

Everything now works as expected. In my opinion these changes can be merged.
Great work!

**Review by sauterl — CHANGES_REQUESTED** [review]
Thanks for the PR! Overall, this looks very good. However, there are a few minor changes requried before we merge this.

Also, please remove the openseadragon images (`src/assets/`), they are loaded elsewhere

File: src/app/core/basics/resolver.service.ts [hunk 1]
```diff
@@ -131,4 +142,39 @@ export class ResolverService {
...
+    let baseUrl = metadata.get('IIIF.resourceUrl')
+    if (!(baseUrl == null || baseUrl.trim().length === 0)) {
+      if (!baseUrl.endsWith('/')) {
+        baseUrl = baseUrl.concat('/')
+      }
+      if (!rawPath) {
+        const xWidth = width ? width : ''
+        const yHeight = height ? height : ''
+        const domain = 'IIIF'
+        const size = (height || width) ? (xWidth + ',' + yHeight) : (metadata.get('IIIF.size') || 'full')
```
[the remark this answers is not in the document]
Comment (singaltanmay) [reply src/app/core/basics/resolver.service.ts]: Done

File: src/app/objectdetails/quick-viewer.component.ts [hunk 2]
```diff
@@ -62,6 +63,41 @@ export class QuickViewerComponent {
...
+  ngAfterViewInit(): void {
+    let url = this.iiifResourceUrl
+    if (!url) {
+      return null
+    } else {
+      url = url + 'info.json';
+    }
+    console.log('ngAfterViewInit URL ', url)
+    console.log('Creating an openseadragon viewer with the url: ', this.iiifResourceUrl)
+    const viewer = openseadragon({
```
[the remark this answers is not in the document]
Comment (singaltanmay) [reply src/app/objectdetails/quick-viewer.component.ts]: Done

**Comment by singaltanmay** [comment]
> Thanks for the PR! Overall, this looks very good. However, there are a few minor changes requried before we merge this.
> 
> Also, please remove the openseadragon images (`src/assets/`), they are loaded elsewhere

Deleting the openseadragon images folder led to the icons not loading so I have left them in for now. I have implemented the other chagnes that you requested

**Review by sauterl — APPROVED** [review]
Thank you for your work. Indeed, the assets have to stay.

**Pull request merged**